web/client/src/utils/index.ts

```diff
@@ -10,18 +10,26 @@ export function isFalseOrNil(value: unknown): boolean {
   return isNil(value) || isFalse(value)
 }
 
-export function isString(value: unknown): boolean {
+export function isString(value: unknown): value is string {
   return typeof value === 'string'
 }
 
+export function isNumber(value: unknown): value is number {
+  return typeof value === 'number'
+}
+
+export function isPrimitive(value: unknown): value is Primitive {
+  return isString(value) || isNumber(value) || typeof value === 'boolean'
```

Comment:
Do you need to do anything to handle symbols?

Reply:
we don't have a use case for Symbol so i'll just remove it from type